lib/elixir/lib/stream/reducers.ex

```diff
@@ -25,18 +25,35 @@ defmodule Stream.Reducers do
     end
   end
 
+  # Commenting out the current version, since I'm not sure
+  # if there should be only one chunk_by macro or more
+  # defmacro chunk_by(callback, fun \\ nil) do
+  #   quote do
+  #     fn
+  #       entry, acc(head, {buffer, value}, tail) ->
+  #         new_value = unquote(callback).(entry)
+  #         if new_value == value do
+  #           skip(acc(head, {[entry | buffer], value}, tail))
+  #         else
+  #           next_with_acc(unquote(fun), :lists.reverse(buffer), head, {[entry], new_value}, tail)
+  #         end
+  #       entry, acc(head, nil, tail) ->
+  #         skip(acc(head, {[entry], unquote(callback).(entry)}, tail))
+  #     end
+  #   end
+  # end
+
   defmacro chunk_by(callback, fun \\ nil) do
     quote do
-      fn
-        entry, acc(head, {buffer, value}, tail) ->
-          new_value = unquote(callback).(entry)
-          if new_value == value do
-            skip(acc(head, {[entry | buffer], value}, tail))
-          else
-            next_with_acc(unquote(fun), :lists.reverse(buffer), head, {[entry], new_value}, tail)
-          end
-        entry, acc(head, nil, tail) ->
-          skip(acc(head, {[entry], unquote(callback).(entry)}, tail))
+      fn(entry, acc(head, {buffer, value}, tail) = acc) ->
```

Comment:
Please don't change the code style. :) Keep it as before with `fn` on its own line and each clause indented two spaces.